data_override/data_override.F90

```diff
@@ -569,120 +134,93 @@ subroutine data_override_unset_domains(unset_Atm, unset_Ocean, &
-      "data_override_unset_domains attempted to work on a Land_domain that had not been set.")
-    lnd_domain = NULL_DOMAIN2D
-    lnd_on = .false.
-    if (allocated(lon_local_lnd)) deallocate(lon_local_lnd)
-    if (allocated(lat_local_lnd)) deallocate(lat_local_lnd)
+  if (present(unset_Land)) then ; if (unset_Land) then
+    select case (lnd_mode)
+      case (r4_kind)
+        call data_override_unset_lnd_r4
+      case (r8_kind)
+        call data_override_unset_lnd_r8
+      case default
+        if (fail_if_not_set) call mpp_error(FATAL, &
+          "data_override_unset_domains: attempted to unset an Land_domain that has not been set.")
+    end select
+    lnd_mode = 0
   endif ; endif
-  if (PRESENT(unset_Ice)) then ; if (unset_Ice) then
-    if (fail_if_not_set .and. .not.ice_on) call mpp_error(FATAL, &
-      "data_override_unset_domains attempted to work on an Ice_domain that had not been set.")
-    ice_domain = NULL_DOMAIN2D
-    ice_on = .false.
-    if (allocated(lon_local_ice)) deallocate(lon_local_ice)
-    if (allocated(lat_local_ice)) deallocate(lat_local_ice)
+  if (present(unset_Ice)) then ; if (unset_Ice) then
+    select case (ice_mode)
+      case (r4_kind)
+        call data_override_unset_ice_r4
+      case (r8_kind)
+        call data_override_unset_ice_r8
+      case default
+        if (fail_if_not_set) call mpp_error(FATAL, &
+          "data_override_unset_domains: attempted to unset an Ice_domain that has not been set.")
+    end select
+    ice_mode = 0
   endif ; endif
-
 end subroutine data_override_unset_domains
 
-!> @brief Given a gridname, this routine returns the working domain associated with this gridname
-subroutine get_domain(gridname, domain, comp_domain)
-  character(len=3), intent(in) :: gridname
-  type(domain2D), intent(inout) :: domain
-  integer, intent(out), optional :: comp_domain(4) !< istart,iend,jstart,jend for compute domain
-
-  domain = NULL_DOMAIN2D
-  select case (gridname)
-     case('OCN')
-        domain = ocn_domain
-     case('ATM')
-        domain = atm_domain
-     case('LND')
-        domain = lnd_domain
-     case('ICE')
-        domain = ice_domain
-     case default
-        call mpp_error(FATAL,'error in data_override get_domain')
-  end select
-  if(domain .EQ. NULL_DOMAIN2D) call mpp_error(FATAL,'data_override: failure in get_domain')
-  if(present(comp_domain)) &
-     call mpp_get_compute_domain(domain,comp_domain(1),comp_domain(2),comp_domain(3),comp_domain(4))
-end subroutine get_domain
-
-!> @brief Given a gridname, this routine returns the working domain associated with this gridname
-subroutine get_domainUG(gridname, UGdomain, comp_domain)
-  character(len=3), intent(in) :: gridname
-  type(domainUG), intent(inout) :: UGdomain
-  integer, intent(out), optional :: comp_domain(4) !< istart,iend,jstart,jend for compute domain
-  type(domain2D), pointer :: SGdomain => NULL()
+!> @brief Routine to perform data override for scalar fields
```

Comment:
This description is inaccurate.  This routine selects the correct routine to call to do the data override for scalar fields.

Reply:
This subroutine was removed in b89c8db621164eef22e0d9220a2046a4a1ceaa98.